### DISCUSSION TITLE
Better Obsidian integration

### DISCUSSION THREAD

**Discussion by xplosionmind** [maximevaillancourt/digital-garden-jekyll-template#79] [Ideas]
The most popular and arguably the best environment to write, edit and manage digital gardens right now is [Obsidian](https://obsidian.md).

Obsidian has a few great functionalities which are lacking in this template, such as:
- [ ] support file embedding, with `![[file-to-embed]]` syntax
- [x] support the ability to give a different name to a wiki link, for example `[[Example note|this is an example]]` ✅ #38 
- [ ] support the ability to link to anchors inside a file, for example `[[Example note#Example Anchor]]`
- [ ] handle the use of the two features above together, for example `[[Example note#Example Anchor|This is an example!]]`
- [x] ~~support note aliases, for example when in the front matter of “beautiful note” there is `aliases: ["nice note", "cute note"]`, writing in a markdown file ``[[nice note]]`` creates a link to “beautiful note”~~. Actually, Obsidian aliases don't work like this.

I saw somewhere in the files that a reference to Obsidian is already present, I believe most of the digital gardeners who are using this template are already using this app.

**Comment by fabriceliut** [comment]
It could be great, I use Obsidian to write my note before publishing to my garden :)

**Comment by maximevaillancourt** [comment]
I just added support for optional labels in #38. Thanks for the suggestion! I may tackle the other suggestions later.

**Comment by redstreet** [comment]
FYI, for anyone who is interested: I was able to get mermaid intergration with [jekyll-spaceship](https://github.com/jeffreytse/jekyll-spaceship#configuration), which Obsidian supports.

**Comment by meewgumi** [comment]
@xplosionmind cc: @fabriceliut @maximevaillancourt 

I worked on a plugin for obsidian image embeds here: https://github.com/meewgumi/green-web-template/blob/main/_plugins/obsidian_images_generator.rb

The trick is setting the priority higher than the bidirectional link generator, so that the images generator runs first. I also set `image_path` to be `assets/images` but if you're just keeping your images/attachments in `assets` you'd have to update that variable

I wrote some notes here on [how I set up Obsidian](https://garden.megu.space/obsidian-setup.html) and [obsidian images generator technical notes here](https://garden.megu.space/wiki-image-link-testing.html). As you can see in the [raw markdown file](https://github.com/meewgumi/green-web-template/blob/main/_notes/Wiki%20Image%20Link%20Testing.md), the plugin can handle both regular MD images and Obsidian formatted image embeds, with or without the proper assets path

Just remember to update your `image_path` in the `.rb` file itself to reflect where your images are being saved! Hope this makes sense to everyone!
